packages/cosmjs/src/core/signingClient/cosmWasmClient.ts

```diff
@@ -0,0 +1,86 @@
+import { CosmWasmClient, SigningCosmWasmClient, SigningCosmWasmClientOptions } from '@cosmjs/cosmwasm-stargate';
+import { OfflineSigner } from '@cosmjs/proto-signing';
+import { createSeiAminoTypes, createSeiRegistry } from './stargateClient';
+
+/**
+ * Returns a interface used to interact with the CosmWASM Contracts on chain.
+ * `getCosmWasmClient` contains helper functions to get a `CosmWasmClient`. Thi should be used for querying smart contracts only. If you need to call a contracts execute msg, see the `SigningCosmWasmClient` below.
+ *
+ * @example Getting a Client
+ * ```tsx
+ * import { DirectSecp256k1HdWallet } from "@cosmjs/proto-signing";
+ * import { getCosmWasmClient } from "@sei-js/cosmjs";
+ *
+ * // Create a CosmWasmClient
+ * const cosmWasmClient = await getCosmWasmClient(RPC_URL);
+ * ```
+ *
+ * @example Querying a Contract
+ * Build the `queryMsg` according to the contracts specific query specifications. Each contract will define its own queryable state, so check the contract documentation for the correct query message format by examining the contract source code or documentation.
+ * ```tsx
+ * // Create the query msg json based on contract specific query specifications
+ * const queryMsg = {
+ *     tokens: {
+ *         owner: address,
+ *     },
+ * };
+ *
+ * // Query a smart contract state
+ * const queryResponse = cosmWasmClient.queryContractSmart(CONTRACT_ADDR, queryMsg);
+ * ```
+ *
+ * @param rpcEndpoint The url of the RPC Endpoint used to connect to the Sei chain.
+ * @returns A CosmWasmClient used to interact with the Sei chain.
+ * @category Clients
+ */
+export const getCosmWasmClient = async (rpcEndpoint: string): Promise<CosmWasmClient> => {
```

Comment:
we should add support for HttpEndpoints here too - https://cosmos.github.io/cosmjs/latest/cosmwasm-stargate/classes/CosmWasmClient.html#connect

they're used to pass in custom headers (like api keys for private rpc nodes)